### ISSUE TITLE
Plugin fails to launch non-steam programs/shortcuts

### ISSUE THREAD

**Issue by rrazgriz** [Garulf/Steam-Search#59]
Entries in the steam library that are non-steam games/programs/shortcuts will show up in search, but will not launch when selected. These non-steam entries should launch directly, and should not offer a context menu with store/news/uninstall options.

This seems to be due to the [`launch_game` method](https://github.com/Garulf/Steam-Search/blob/main/plugin/main.py#L70-L71) using the URI to launch game entries, which fails for non-steam game shortcuts.

Non-steam game shortcuts (on my system) are stored in the file `C:\Program Files (x86)\Steam\userdata\59574360\config\shortcuts.vdf`, as a binary key-value store. I was able to read this file with the [`vdf`](https://github.com/ValvePython/vdf) library to extract the data for each entry's exe, launch directory, and launch options.

**Comment by Garulf** [comment]
That's very strange. The URI should work for all Steam library entries. 

If you add a Non-Steam game to desktop does that shortcut work for you?

**Comment by rrazgriz** [comment]
Yes, it does seem to work. Adding 7-zip as a non-steam game doesn't launch through the flow launcher plugin, but when adding a desktop shortcut, it work (results in this uri): 
![image](https://github.com/Garulf/Steam-Search/assets/47901762/4af75d27-c801-46fa-ab97-c04ffcc214d7)

I get the following flow launcher log entry whenever attempting to launch an entry like this: 
```22:42:33.8261-04:00 - WARN  - JsonRPCPlugin.ExecuteAsync - Empty JSON-RPC Response.```

Here's my [shortcuts.vdf](https://github.com/Garulf/Steam-Search/files/11997262/shortcuts.vdf.zip) for reference.

**Comment by Garulf** [comment]
Sorry could you please post the full error log?

**Comment by rrazgriz** [comment]
Sorry for the delay, lost track of this.

Here's a log consisting exclusively of the application starting up, and trying to run a non-steam shortcut: [2023-08-02.txt](https://github.com/Garulf/Steam-Search/files/12246703/2023-08-02.txt)

`23:55:28.0596-04:00` is the timestamp at which I try to run a non-steam shortcut.

**Comment by zurce** [comment]
This still an issue in 2025, anything i can provide to help fixing it?

![Image](https://github.com/user-attachments/assets/11bebee3-b0df-4775-b681-6dbabaaae78b)

**Comment by Garulf** [comment]
Related issue here:

https://github.com/Garulf/Steam-Search/issues/57#issuecomment-4955178712
